lib/server_health_check.rb

```diff
@@ -25,13 +25,17 @@ def redis!(host: nil, port: 6379)
 
   def active_record!
     begin
-      ActiveRecord::Base.connection
-      @results[:active_record] = OK
-      true
+      if ActiveRecord::Base.connection_pool.with_connection { |connection| connection.active? }
+        @results[:active_record] = OK
+        true
+      else
+        @results[:active_record] = "Failed: unable to connect to database"
+        false
+      end
     rescue NameError
```

Comment:
I think the idea behind the rescue is to guard against using the check when you aren't actually using `ActiveRecord`. Since `NameError` extends `StandardError`, it has to be a separate rescue, unless you want to add a conditional within the `StandardError` rescue (which I think would be not as clean).

I feel like a line to draw out the `NameError` above the begin would look awkward and require a comment to be clear, so I feel like keeping it within the begin block is better.

With regards to returning like normal, I think propagating a `NameError` would make diagnosing using the check when `ActiveRecord` isn't used much easier than returning with a different error message (or worse, hiding behind the same message as a `StandardError`).

Thus, personally, I think leaving the `NameError` rescue as is would be best.